test/test_tensor.py

```diff
@@ -154,5 +155,34 @@ def test_random_fns_are_deterministic_with_seed(self):
         b = random_fn(10,10).realize()
         np.testing.assert_allclose(a.numpy(), b.numpy())
 
+  def test_zeros_like_has_same_dtype(self):
+    a = Tensor([1, 2, 3])
+    b = Tensor.zeros_like(a)
+    self.assertTrue(a.dtype == b.dtype)
+
+    a = Tensor([1, 2, 3], dtype=dtypes.float16)
+    b = Tensor.zeros_like(a)
+    self.assertTrue(a.dtype == b.dtype)
```

Comment:
I'd say just 2 cases is enough (default and say int8), you can add `assert` to check shapes are the same

Reply:
@python273 updated the test, let me know if there's a better way of doing this